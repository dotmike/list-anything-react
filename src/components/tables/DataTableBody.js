import React from "react";
import PropTypes from "prop-types";
import { Table } from "semantic-ui-react";

const styles = {
  body: { cursor: "pointer" }
};

const DataTableBody = ({ renderBody }) => (
  <Table.Body style={styles.body}>{renderBody()}</Table.Body>
);

DataTableBody.propTypes = {
  // ownProps
  renderBody: PropTypes.func.isRequired
};

export default DataTableBody;
