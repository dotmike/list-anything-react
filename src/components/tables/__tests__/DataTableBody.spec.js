import React from "react";
import { shallow } from "enzyme";

import DataTableBody from "../DataTableBody";

const defaultProps = {
  renderBody: () => <div>Testando...</div>
};

describe("DataTableBody", () => {
  it("renders correctly", () => {
    defaultProps.results = [];

    const wrapper = shallow(<DataTableBody {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
