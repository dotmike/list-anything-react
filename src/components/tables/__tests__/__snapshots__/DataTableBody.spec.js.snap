// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableBody renders correctly 1`] = `
<TableBody
  as="tbody"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div>
    Testando...
  </div>
</TableBody>
`;
